src/mlang/backend_compilers/bir_to_ocaml.ml

```diff
@@ -0,0 +1,387 @@
+  Format.fprintf oc "@[<v 1>let m_%s_%s (context : m_context) : unit =@,%a@]"
+    tname (Pos.unmark rov.rov_name) (generate_stmts program) stmts
+
+let generate_rov_functions (oc : Format.formatter) (program : Bir.program) :
+    unit =
+  let rovs = Bir.ROVMap.bindings program.rules_and_verifs in
+  let _, rovs = List.split rovs in
+  Format.pp_print_list ~pp_sep:pp_function_separator
+    (generate_rov_function program)
+    oc rovs
+
+let generate_header (oc : Format.formatter) () : unit =
+  Format.fprintf oc "@[<v 0>open Mvalue@,@]"
+
+let generate_output (oc : Format.formatter)
+    (function_spec : Bir_interface.bir_function) : unit =
+  let output_vars =
+    List.map fst (Bir.VariableMap.bindings function_spec.func_outputs)
+  in
+  let name_and_pos_list var_list =
+    List.map
+      (fun var -> (get_var_pos var, Pos.unmark var.mir_var.name))
+      var_list
+  in
+  let print_line fmt (position, name) =
+    Format.fprintf fmt "(\"%s\", %d)" name position
+  in
+  let pp_print_output_get fmt output_vars =
+    Format.pp_print_list ~pp_sep:pp_statement_separator print_line fmt
+      (name_and_pos_list output_vars)
+  in
+  Format.fprintf oc
+    "let output (tgv : m_array) : output_array =@,\
+     let rev_code_from_position (x , p) : revenue_code = {alias = x; value = \
+     (Array.get tgv p).value} in@,\
+     let output_positions_array = @[<v>[|%a|]@] in@,\
+     Array.map rev_code_from_position output_positions_array"
+    pp_print_output_get output_vars
+
+let generate_input_handler (oc : Format.formatter)
+    (function_spec : Bir_interface.bir_function) : unit =
+  let input_vars =
+    List.map fst (Bir.VariableMap.bindings function_spec.func_variable_inputs)
+  in
+  let get_position_and_alias variable : (int * string) option =
+    let tgv_pos = get_var_pos variable in
+    Option.map (fun alias -> (tgv_pos, alias)) (get_var_alias variable)
+  in
+  let pp_print_line fmt variable : unit =
+    match get_position_and_alias variable with
+    | Some (position, alias) ->
+        Format.fprintf fmt "(\"%s\", %d);@," alias position
+    | None -> ()
+  in
+  let pp_print_position_map fmt input_vars =
+    Format.pp_print_list ~pp_sep:pp_empty_separator pp_print_line fmt input_vars
+  in
+  Format.fprintf oc
+    "let input_handler (tgv : m_array) (entry_list : revenue_code list) : unit \
+     =@,\
+     let entry_positions_array = @[<v>[|%a|]@] in@,\
+     let entry_positions_map = @,\
+     Array.fold_left (fun pos_map (alias, position) -> TgvPositionMap.add \
+     alias position pos_map)@,\
+     TgvPositionMap.empty entry_positions_array in@,\
+     let init_tgv_var (entry_var : revenue_code) : unit =@,\
+     Array.set tgv (TgvPositionMap.find entry_var.alias entry_positions_map)@,\
+     {undefined = false ; value = entry_var.value} in@,\
+     List.iter init_tgv_var entry_list" pp_print_position_map input_vars
+(* Prévoir les cas : variable manquante, variable en trop dans entry_list,
+   variable définie n fois*)
```

Comment:
Is it done ?

Reply:
Good catch. No it isn't. Seems like all tests are kindly providing well built entry lists. Not so obvious a check to add, though.

Reply:
Don't bother, just put a comment stating we don't check those cases.